lib/migrate/extractTextPlugin/extractTextPlugin.js

```diff
@@ -1,58 +1,111 @@
-			.size() > 0;
-	return found;
-}
-
-/**
- *
- * Transform for ExtractTextPlugin arguments. Consolidates arguments into single options object.
+ * Transform for ExtractTextPlugin. Replaces it with MiniCssExtractPlugin and translates
+ * configuration where possible.
  *
  * @param {Object} j - jscodeshift top-level import
  * @param {Node} ast - jscodeshift ast to transform
  * @returns {Node} ast - jscodeshift ast
  */
+module.exports = function transformExtractTextPlugin(j, ast) {
+	const devLoaders = [];
+	const prodLoaders = [];
+
+	let elements;
+	let loaderIndex;
+	let loaderPath;
+	let newElements;
+	let pluginExists;
+	let requireExists;
+
+	const filename = getPluginOptionValue(j, ast, "ExtractTextPlugin", "filename");
+	const publicPath = getLoaderOptionValue(j, ast, "ExtractTextPlugin", "publicPath");
+	const currentFallbackLoader = getLoaderOptionValue(j, ast, "ExtractTextPlugin", "fallback");
+	const currentLoader = getLoaderOptionValue(j, ast, "ExtractTextPlugin", "use");
 
-module.exports = function(j, ast) {
-	const changeArguments = function(p) {
-		const args = p.value.arguments;
-
-		const literalArgs = args.filter(p => utils.isType(p, "Literal"));
-		if (literalArgs && literalArgs.length > 1) {
-			const newArgs = j.objectExpression(
-				literalArgs.map((p, index) =>
-					utils.createProperty(j, index === 0 ? "fallback" : "use", p.value)
-				)
-			);
-			p.value.arguments = [newArgs];
+	const loaderOptions = filename || publicPath ? {} : null;
+	filename && Object.assign(loaderOptions, { filename });
+	publicPath && Object.assign(loaderOptions, { publicPath: publicPath.value });
+
+	// Remove require for old plugin
+	requireExists = removeRequire(j, ast, "ExtractTextPlugin");
+
+	// Add require for new plugin
+	requireExists && addRequire(j, ast, "MiniCssExtractPlugin", "mini-css-extract-plugin");
+
+	// Remove old plugin
+	pluginExists = removePluginByName(j, ast, "ExtractTextPlugin");
+
+	// Add new plugin
+	pluginExists && createPluginByName(j, ast, "MiniCssExtractPlugin", loaderOptions);
+
+	// Remove old loader
+	ast.find(j.Identifier, { name: "ExtractTextPlugin" })
+		.filter(path => safeTraverse(path, ["parent", "value"]))
+		.filter(path => isType(path.parent.value, "MemberExpression"))
+		.forEach(path => {
+			loaderPath = safeTraverse(path, ["parent", "parent", "parent", "value"]);
+			loaderPath.elements && (elements = [...loaderPath.elements]);
+			j(path.parent.parent).remove();
+		});
+
+	// Build loader arrays
+	currentLoader && pathToArray(currentLoader, switchedLoader => {
+		const loader = j.memberExpression(j.identifier("MiniCssExtractPlugin"), j.identifier("loader"), false);
+		prodLoaders.push(loader, ...switchedLoader);
+		currentFallbackLoader && pathToArray(currentFallbackLoader, switchedFallbackLoader => {
+			devLoaders.push(...switchedFallbackLoader, ...switchedLoader);
+		});
+	});
+
+	// Add new loader
+	const loader = currentFallbackLoader ?
+		j.conditionalExpression(
```

Comment:
It might be too specific, is there another way of adding a loader?

Reply:
I created an `addConditionalLoader` utility function per your comment above. I think it'll be useful in the long run, as it makes it easy to add loaders based on `NODE_ENV`.